core/lib/vm/src/oracles/storage.rs

```diff
@@ -201,7 +215,11 @@ impl<S: WriteStorage, H: HistoryMode> StorageOracle<S, H> {
     fn base_price_for_write_query(&self, query: &LogQuery) -> u32 {
```

Comment:
(It is not for this line, but a nit for lines that are not in this PR)

Since you've already added the `set_initial_value` function, you can replace the following snippet:

```
if !self.initial_values.inner().contains_key(&key) {
            self.initial_values
                .insert(key, current_value, query.timestamp);
        }
```

with it.